sonar-kotlin-gradle/src/main/java/org/sonarsource/kotlin/gradle/checks/RootProjectNamePresentCheck.kt

```diff
@@ -0,0 +1,76 @@
+/*
+ * SonarSource Kotlin
+ * Copyright (C) 2018-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonarsource.kotlin.gradle.checks
+
+import org.jetbrains.kotlin.lexer.KtTokens
+import org.jetbrains.kotlin.psi.KtBinaryExpression
+import org.jetbrains.kotlin.psi.KtDotQualifiedExpression
+import org.jetbrains.kotlin.psi.KtFile
+import org.jetbrains.kotlin.psi.KtNameReferenceExpression
+import org.jetbrains.kotlin.psi.KtTreeVisitorVoid
+import org.jetbrains.kotlin.psi.psiUtil.startOffset
+import org.sonar.check.Rule
+import org.sonarsource.kotlin.api.checks.AbstractCheck
+import org.sonarsource.kotlin.api.frontend.KotlinFileContext
+import org.sonarsource.kotlin.api.reporting.KotlinTextRanges.textRange
+import org.sonarsource.kotlin.api.reporting.message
+
+private const val SETTINGS_FILE_NAME = "settings.gradle.kts"
+private const val IDENTIFIER_ROOT_PROJECT = "rootProject"
+private const val IDENTIFIER_NAME = "name"
+
+@Rule(key = "S6625")
+class RootProjectNamePresentCheck : AbstractCheck() {
+
+    override fun visitKtFile(file: KtFile, kotlinFileContext: KotlinFileContext) {
+        if (!file.name.endsWith(SETTINGS_FILE_NAME)) return
+
+        val visitor = ScanRootProjectNameAssign()
+        file.acceptChildren(visitor)
+        if (!visitor.isRootProjectNameAssigned) {
+
+            kotlinFileContext.reportIssue(
+                kotlinFileContext.textRange(file.startOffset, file.startOffset + 1),
```

Comment:
I've just realised that if file is empty this might fail. Maybe let's report on a file? you can pass `null` as a text range. Look at `FileHeaderCheck`

Reply:
Cool :-) Fixed.